modules/rules/ipg.js

```diff
@@ -1,143 +1,266 @@
-
-                const penalty = $(".alert-warning").first();
-                if (penalty.length > 0) {
-                    result.push("*" + penalty.text().replace(/\r?\n|\r/g, ' ').trim().replace(" ", ": ") + "*\n");
-                    penalty.remove();
-                }
-
-                const article = $("article");
-                article.find(".entry-header, .page-navigation, em").remove();
-                const that = this;
-                const foundTopics = [];
-                if(parameters.length === 1){
-                    article.find("td").each((index,element)=>{
-                        result.push($(element).text()+"\n");
-                    });
-                    article.find("h2, p, li, .card-header").each((index,element)=>{
-                        let text = $(element).text();
-                        const nodeName = $(element).prop("nodeName").toLowerCase();
-                        if(this.handlers[text.trim()]){
-                            foundTopics.push(text);
-                            text = "\n**" + text + "**";
-                        }
-
-                        if(result.join("").length<that.maxPreview){
-                            result.push(text+"\n");
-                        }
-                    });
-                }else{
-                    const topics = parameters.slice(1,this.maxTopics+1);
-                    let writeToggle = false;
-                    article.find("h2, p, li, .card-header").each((index,element) => {
-                        let text = $(element).text();
-                        const nodeName = $(element).prop("nodeName").toLowerCase();
-                        if(this.handlers[text.trim()]){
-                            writeToggle = topics.includes(this.handlers[text.trim()]);
-                            foundTopics.push(text);
-                            text = "\n**" + text + "**\n";
-                        }else{
-                            text = text +"\n";
-                        }
-                        if(!writeToggle){
-                            return;
-                        }
-                        result.push(text);
-                    });
-                }
-                result.push("\n**Available topics:** *"+foundTopics.join("*, *")+"*\n");
-                return msg.channel.sendMessage(result.join("").replace(/\n\s*\n\s*\n/g, '\n\n'),{"split":"true"});
+    prettyPrintSectionTitle(section) {
+        return this.titleCase(section.replace("-", " "));
+    }
+
+    formatSections(sections) {
+        return sections.map(s => "*" + s + "*").join(", ");
+    }
+
+    formatPreview(entry) {
+        const previewText = _.truncate(entry.text, {length: this.maxPreview, separator: ' '});
+        return  `**${entry.title}**\n${previewText}`;
+    }
+
+    formatChapterEntry(entry) {
+        const text = entry.text || this.formatPreview(this.ipgData[entry.sections[0]]);
+
+        return `**IPG ${entry.title}**
```

Comment:
Yes, because it messes with the indentation. You'll either indent it to have it match the rest of the code (and then end up with extra spaces in your message) or not indent it to send the right amount of spaces, but end up with messy looking code. That's why I prefer either the line break-joins or simply using `\n` instead of a literal line break.